xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VdpauGL.cpp

```diff
@@ -25,20 +24,32 @@ bool CInteropState::Init(void *device, void *procFunc, int64_t ident)
   m_procFunc = procFunc;
   m_ident = ident;
 
-  m_interop.glVDPAUInitNV = (PFNGLVDPAUINITNVPROC)glXGetProcAddress((const GLubyte *) "glVDPAUInitNV");
-  m_interop.glVDPAUFiniNV = (PFNGLVDPAUFININVPROC)glXGetProcAddress((const GLubyte *) "glVDPAUFiniNV");
-  m_interop.glVDPAURegisterOutputSurfaceNV = (PFNGLVDPAUREGISTEROUTPUTSURFACENVPROC)glXGetProcAddress((const GLubyte *) "glVDPAURegisterOutputSurfaceNV");
-  m_interop.glVDPAURegisterVideoSurfaceNV = (PFNGLVDPAUREGISTERVIDEOSURFACENVPROC)glXGetProcAddress((const GLubyte *) "glVDPAURegisterVideoSurfaceNV");
-  m_interop.glVDPAUIsSurfaceNV = (PFNGLVDPAUISSURFACENVPROC)glXGetProcAddress((const GLubyte *) "glVDPAUIsSurfaceNV");
-  m_interop.glVDPAUUnregisterSurfaceNV = (PFNGLVDPAUUNREGISTERSURFACENVPROC)glXGetProcAddress((const GLubyte *) "glVDPAUUnregisterSurfaceNV");
-  m_interop.glVDPAUSurfaceAccessNV = (PFNGLVDPAUSURFACEACCESSNVPROC)glXGetProcAddress((const GLubyte *) "glVDPAUSurfaceAccessNV");
-  m_interop.glVDPAUMapSurfacesNV = (PFNGLVDPAUMAPSURFACESNVPROC)glXGetProcAddress((const GLubyte *) "glVDPAUMapSurfacesNV");
-  m_interop.glVDPAUUnmapSurfacesNV = (PFNGLVDPAUUNMAPSURFACESNVPROC)glXGetProcAddress((const GLubyte *) "glVDPAUUnmapSurfacesNV");
-  m_interop.glVDPAUGetSurfaceivNV = (PFNGLVDPAUGETSURFACEIVNVPROC)glXGetProcAddress((const GLubyte *) "glVDPAUGetSurfaceivNV");
-
-  while (glGetError() != GL_NO_ERROR);
+  m_interop.glVDPAUInitNV =
+      (PFNGLVDPAUINITNVPROC)glXGetProcAddress((const GLubyte*)"glVDPAUInitNV");
+  m_interop.glVDPAUFiniNV =
+      (PFNGLVDPAUFININVPROC)glXGetProcAddress((const GLubyte*)"glVDPAUFiniNV");
+  m_interop.glVDPAURegisterOutputSurfaceNV =
+      (PFNGLVDPAUREGISTEROUTPUTSURFACENVPROC)glXGetProcAddress(
+          (const GLubyte*)"glVDPAURegisterOutputSurfaceNV");
+  m_interop.glVDPAURegisterVideoSurfaceNV = (PFNGLVDPAUREGISTERVIDEOSURFACENVPROC)glXGetProcAddress(
+      (const GLubyte*)"glVDPAURegisterVideoSurfaceNV");
+  m_interop.glVDPAUIsSurfaceNV =
+      (PFNGLVDPAUISSURFACENVPROC)glXGetProcAddress((const GLubyte*)"glVDPAUIsSurfaceNV");
+  m_interop.glVDPAUUnregisterSurfaceNV = (PFNGLVDPAUUNREGISTERSURFACENVPROC)glXGetProcAddress(
+      (const GLubyte*)"glVDPAUUnregisterSurfaceNV");
+  m_interop.glVDPAUSurfaceAccessNV =
+      (PFNGLVDPAUSURFACEACCESSNVPROC)glXGetProcAddress((const GLubyte*)"glVDPAUSurfaceAccessNV");
+  m_interop.glVDPAUMapSurfacesNV =
+      (PFNGLVDPAUMAPSURFACESNVPROC)glXGetProcAddress((const GLubyte*)"glVDPAUMapSurfacesNV");
+  m_interop.glVDPAUUnmapSurfacesNV =
+      (PFNGLVDPAUUNMAPSURFACESNVPROC)glXGetProcAddress((const GLubyte*)"glVDPAUUnmapSurfacesNV");
+  m_interop.glVDPAUGetSurfaceivNV =
+      (PFNGLVDPAUGETSURFACEIVNVPROC)glXGetProcAddress((const GLubyte*)"glVDPAUGetSurfaceivNV");
```

Comment:
Is this function pointer only called when GLX is used?

If not, maybe it would be interesting to set a function pointer in the class (here or separate source files) which sets to glXGetProcAddress or alternatively to eglGetProcAddress.

Furthermore maybe an extra function that sets the pointers, maybe:
```cpp
bool SetGLProcAddress(void*&  address, const std::string& funcname)
{
  address = glXGetProcAddress(reinterpret_cast<GLubyte*>(funcname.c_str())); // Maybe here a stored function pointer instead glXGetProcAddress to allow also eglGetProcAddress.
  return (address != nullptr && gl::GetError() == GL_NO_ERROR)
}

// Here to use
if (!SetGLProcAddress(m_interop.glVDPAUGetSurfaceivNV, "glVDPAUGetSurfaceivNV"))
  break;
```

That way, clang could stay a little quieter and if there is such a function in between, you could switch between GLX, EGL and possibly even WGL.

Would be interesting for here or in another request.

I will look through the request again this evening, but I have to say it again, to have this request in there would be awesome.

Reply:
VDPAU is only used when glx is used.

I intend to do some cleanup of this later because we use getProcAddress methods in a few various places and it would be nice to have uniform access to this.

For now the existing code still works 😀